pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -347,9 +347,21 @@ func (r *ReconcileToolchainStatus) membersHandleStatus(logger logr.Logger, toolc
 
 	// add member cluster statuses to toolchainstatus
 	ready = compareAndAssignMemberStatuses(logger, toolchainStatus, members) && ready
+	// assign apiEndpoint to Members
+	assignApiEndpoint(toolchainStatus,memberClusters)
 	return ready
 }
 
+func assignApiEndpoint(toolchainStatus *toolchainv1alpha1.ToolchainStatus, memberClusters []*cluster.CachedToolchainCluster) {
+	for _, memberCluster := range memberClusters {
+		for index, member := range toolchainStatus.Status.Members{
+			if member.ClusterName == memberCluster.Name {
+				toolchainStatus.Status.Members[index].ApiEndpoint = memberCluster.APIEndpoint
+				break
+			}
+		}
```

Comment:
It's just a proposal, but do you think that we could move this part into `compareAndAssignMemberStatuses` method and do the assignment there? https://github.com/codeready-toolchain/host-operator/blob/57ec214efcf9269277c617dfc714784fb6edc862/pkg/controller/toolchainstatus/toolchainstatus_controller.go#L424-L431

It's pretty hard to follow the association and the source of names of the member clusters

Reply:
i was worried that it doesn't add to the confusion to move to this function, considering the function name says to update memberStatuses, while apiEndpoint is in Member

Reply:
You are right, it's probably more appropriate at the place where you put it. In addition, the code that I referenced is only for the creation of the new Members that are not part of the status yet and doesn't update of the already existing ones. Your solution will update both of them. :+1: